src/parser.cpp

```diff
@@ -248,7 +248,7 @@ namespace ada::parser {
           // While c is an ASCII digit:
           while (unicode::is_ascii_digit(*pointer)) {
             // Let number be c interpreted as decimal number.
-            uint16_t number = static_cast<uint16_t>(*pointer);
+            auto number = static_cast<uint16_t>(std::strtoul(pointer, nullptr, 10));
```

Comment:
- I'm still not sure if the assumption that `std::string_view::iterator` is `const char*` is portable
- `input` is not guaranteed to be null-terminated
- `strtoul` has a number of behaviors that are probably incompatible with URL parsing (e.g. support for hex or octal digits, leading whitespace, etc.)
- You probably don't want to truncate the result here by casting it to `uint16_t` without regard for overflow

Reply:
Since pointer points to a ascii_digit, I think the best way to implement this is: `int number = *pointer - '0';`

Reply:
Since the pointer points to an ascii_digit, I think the best way to implement this is: `int number = *pointer - '0';`